### ISSUE TITLE
Pylint 3.0.1 | invalid-field-call | false positive (another)

### ISSUE THREAD

**Issue by mcowpert** [pylint-dev/pylint#9161]
### Bug description

```python
Pylint flags this usage:


    dataclasses.make_dataclass(
        cls_name="nombre",
        fields=[(id, int, dataclasses.field(default=None)) for id in fields],
    )
```
```


### Configuration

_No response_

### Command used

```shell
Invoked via VSCode's Pylint extension
```

### Pylint output

```python
exact output unknown
VSCode displays error message in popup:

> Invalid usage of field, it should be used within a dataclass or the make_dataclass() function. Pylint(E3701:invalid-field-call)
```

### Expected behavior

No error! I believe the code is valid, and it runs with the expected outcome.

### Pylint version

```shell
VSCode Pylint extension panel displays "pylint=3.0.1"
```

### OS / Environment

Ubuntu 22.04
